File: frontend/src/types/order.ts
export type OrderSide = 'BUY' | 'SELL';
export type OrderStatus = 'NEW' | 'FILLED' | 'CANCELLED';

export interface Order {
  id: string;
  symbol: string;
  side: OrderSide;
  quantity: number;
  price: number;
  status: OrderStatus;
  createdAt: string;
}

export interface CreateOrderRequest {
  symbol: string;
  side: OrderSide;
  quantity: number;
  price: number;
}

export interface OrderSummary {
  totalOrders: number;
  openOrders: number;
  cancelledOrders: number;
  filledOrders: number;
  notionalOpenValue: number;
}

export type OrderFilter = 'ALL' | OrderStatus;

File: backend/src/main/java/com/example/oms/model/Order.java
package com.example.oms.model;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record Order(
        UUID id,
        String symbol,
        OrderSide side,
        int quantity,
        BigDecimal price,
        OrderStatus status,
        Instant createdAt
) {
}
